pkg/skaffold/schema/profiles.go

```diff
@@ -55,11 +59,36 @@ func ApplyProfiles(c *latest.SkaffoldConfig, opts cfg.SkaffoldOptions) error {
 		}
 	}
 
+	return enableEffectiveKubecontext(isContextSpecific, opts.KubeContext, c.Deploy.KubeContext)
```

Comment:
I am really confused why a profile application needs to change the kubecontext. Isn't it the other way around? Profiles are _activated_ if the current context matches the activation condition...

Reply:
Well, both ways are needed. Profiles activated _by_ a kubecontext is what Skaffold already supports. Changing the active kubecontext from a profile is new. This is one of the major use-cases for having the setting in `skaffold.yaml`. For example, if a user as a profile for the `staging` environment, she can determine that a staging kubecontext should be used in that case.
We also discussed about edge cases when multiple profiles try to define conflicting kubecontexts. In that case, Skaffold will stop with an error.

Also see https://github.com/GoogleContainerTools/skaffold/blob/master/docs/design_proposals/configurable-kubecontext.md#option-in-skaffoldyaml (the wording is not very concise though :/)

Reply:
I see!!! 
This is still hard to understand, and it is still surprising to see kubecontext modification inside `ApplyProfiles`. I would rename this method to `checkKubeContextConsistency`. In the runner - where we call `ApplyProfiles` from, we could call `UseKubeContext(c.Deploy.KubeContext)`
Also, I would remove the call to `UseKubeContext` from the cmd and just have it in that one place in the `Runner` after the profiles are applied. WDYT?

Reply:
I agree that it will be easier to follow, if all the `UseKubeContext` calls are made from the same place. Inside from `ApplyProfiles` is not a good way, I'll change that.

On the other hand, `UseKubeContext(opts.KubeContext)` needs to be called before profile resolution, because the CLI flag `--kube-context` should also affect which profiles are activated if they specify a kube-context activation. Especially considering that the next step will also consider kube-context settings from the global config, I think it is easier to call `UseKubeContext` twice.